### ISSUE TITLE
nc_put_vars_double fails in parallel if using netcdf-4 collective

### ISSUE THREAD

**Issue by gsjaardema** [Unidata/netcdf-c#447]
Version 4.5.1-devel.

If I call nc_put_vars_double with stride=1 in parallel with some processors having no data to write, then the `H5Dwrite` call will fail.

The problem is due to the 
```
   if(nels == 0)
      return NC_NOERR; /* cannot write anything */
```
at line 244 of libdispatch/dvarput.c.   If I remove that early return and if stride == 1, then the code will complete correctly.  If that line is left as is, then some processors return early and the code hangs down below `H5Dwrite` due to hdf5 calling `PMPI_Allreduce` if using collective io.

There is another issue if stride != 1, but I will report that in a separate issue.

**Comment by gsjaardema** [comment]
The related issue when `stride != 1` is #448

**Comment by edhartnett** [comment]
@gsjaardema Is this still and issue or should it be closed?

**Comment by gsjaardema** [comment]
This can be closed.

**Comment by edhartnett** [comment]
You have to close it, I can't. ;-)
